### PR TITLE
Build blokus game

### PR DESCRIPTION
**Pull request by yuyuranium**
Built the basics of the blokus game
Can dump debug output using
```
make DEBUG+='-DDEBUG=1'
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+HEADERS := $(wildcard $(INC_DIR)/*.h)
+SOURCES := $(wildcard $(SRC_DIR)/*.c)
+OBJECTS := $(SOURCES:$(SRC_DIR)/%.c=$(BUILD_DIR)/%.o)
+
+__dir := $(shell mkdir -p $(BUILD_DIR))
+
+DEBUG = -g
+CFLAGS = -Wall -I$(INC_DIR) $(DEBUG)
+LDFLAGS = -lpthread
```
Comment (choucl) [inline Makefile:15]: We could add `-lncursesw` here

File: src/blokus.c [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+    for (int i = 0; i < TILE[tile->shape].blk_cnt; ++i) {
+        int y = pos.y + tile->blks[i].y, x = pos.x + tile->blks[i].x;
+        int k = N_ROW * y + x;
+        gcb->map[y][x] = p;
+        gcb->next_empty[gcb->prev_empty[k]] = gcb->next_empty[k];
+        gcb->prev_empty[gcb->next_empty[k]] = gcb->prev_empty[k];
+    }
+
+    encode_tile(gcb->code, tile);  // record the last update with gcb->code
+    gcb->score[p] += TILE[tile->shape].blk_cnt;
```
Comment (choucl) [inline src/blokus.c:465]: We should set the tile on user hand to `NULL` if they place the tile correctly

File: inc/blokus.h [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+typedef struct {
+    int turn;
+    shape_t sel_shape;
+    tile_t *hand[2][SHAPE_Z + 1];
+    int score[2];
+    int next_empty[N_ROW * N_COL];
+    int prev_empty[N_ROW * N_COL];
+    int map[N_ROW][N_COL];
+    char code[16];
```
Comment (choucl) [inline inc/blokus.h:65]: We may add error code attribute in `gcb` so that I can give user corresponding messages

File: Makefile (hunk 1 above)
Comment (yuyuranium) [reply Makefile:15]: Sure! I will refer to yours and add the flag. Thanks.

File: src/blokus.c (hunk 2 above)
Comment (yuyuranium) [reply src/blokus.c:465]: OK! Please wait for my update.

File: inc/blokus.h (hunk 3 above)
Comment (yuyuranium) [reply inc/blokus.h:65]: I think I will include this in another feature PR. Do you think this is OK?

Comment (choucl) [reply inc/blokus.h:65]: That'll be great, thanks

**Pull request merged**